### DISCUSSION TITLE
Outreachy Discussions - 2023 Round 1

### DISCUSSION THREAD

**Discussion by MridulS** [networkx/networkx#6442] [General]
Hello all!

Please use this discussion page to discuss everything for Outreachy 2023 round 1 (May 2023 - August 2023).

If you have any questions or doubts about the processes please put in a reply to this discussion post.

As a first step: applicants who haven't used networkx before should first [install networkx](https://networkx.org/documentation/latest/install.html) on their machines and follow our [introductory tutorial](https://networkx.org/documentation/latest/tutorial.html). This will give applicants a better overview of our codebase and API, which will help the applicant create a strong application. If something isn't clear from the tutorial please raise an issue/put a comment here so we can do a better job of maintaining our tutorial.

Once applicants are comfortable with using networkx, they should go through our [Contributor Guide](https://networkx.org/documentation/latest/developer/contribute.html) to setup our local development environment for networkx.

We have created another repository https://github.com/networkx/outreachy which will have explicit tasks for Outreachy applicants during the contribution period.

To participate with NetworkX as an Outreachy intern applicants are expected to go through multiple tasks in that repository. All tasks are presented as [issues](https://github.com/networkx/outreachy/issues). We will add a couple of more tasks to this repository.

Everyone is expected to start with https://github.com/networkx/outreachy/issues/2.

The applicant should also go through the [New Contributor FAQs](https://networkx.org/documentation/latest/developer/new_contributor_faq.html) as it may answer some of their questions.

As the project will deal with creating new notebooks for our NX-guides please do go through current notebooks at https://networkx.org/nx-guides/.

If the applicant is interested in working on specific issues they can comment on them. Applicants can also pick any [open issue](https://github.com/networkx/networkx/issues) if they like or even open a new issue/discussion if there is something in our documentation or codebase that doesn't make sense :)

Feel free to post here if you have any other questions or issues!

Excited for the coming winter! :)

**Comment by chinmaychahar** [comment]
Hi there! I am Chinmay an outreachy applicant and I'm excited to contribute to NetworkX and collaborate with the community to make meaningful improvements to the project.

**Comment by BhaswatiRoy** [comment]
Hello everyone
I am Bhaswati Roy, a prefinal year CSE student interested in the field of Machine Learning and Data Analysis.
My initial Outreachy application is approved and I am interested in the project "Create Pedagogical Interactive Notebooks for Algorithms Implemented in NetworkX".
Looking forward to contribute to NetworkX

**Comment by devotuoma** [comment]
Hello everyone. Am Erick Otuoma from Nairobi Kenya my initial application approval message arrived late in the night so am eager to contribute to this project and collaborate with everyone to ensure success and timely delivery. Can someone help me to know how I can get to start over please.

**Reply by Daviskojob** [reply]
Welcome 🙏

**Comment by sirichandana-v** [comment]
Hello all! This is Siri Chandana, an aspiring software engineer. My initial application of Outreachy is approved. I'm so excited to start this new journey of learning and contributing to NetworkX.

**Reply by Daviskojob** [reply]
Welcome 🙏

**Comment by Ruchi-here** [comment]
hey @MridulS , are we supposed to pick up the issues from main codebase (networkx/networkx) or just need to resolve the issues that are given in networkx/outreachy ? can you please explain

**Reply by paulitapb** [reply]
You need to go through the issues in the network/outreachy first and then start making contributions in the main repo. Some of the issues in the other repo are intended to check your setup configuration and introduce you to networkx codebase and documentation so you can be ready to start solving issues in the main repo.

**Comment by Ruchi-here** [comment]
Because its written here

![image](https://user-images.githubusercontent.com/122676573/223464700-a30174b0-0c74-4815-958b-c6e228bd771d.png)

**Comment by akanshajais** [comment]
Hi @MridulS  I am Akansha a final year CS student having well versed knowledge of **Python Programming , Git , Github , Jupyter Notebook and Scientific Data Analysis**.I have been approved for contribution round of outreachy internship program and I am eager to start my contribution to NetworkX project with my expertise. I am looking for your guidance to successfully complete my contribution.

**Reply by Daviskojob** [reply]
Welcome

**Comment by Sreeyusha** [comment]
Hi @MridulS, I am Sreeyusha from India. I am an Outreachy applicant approved for the contribution period. I am excited to contribute to this NetworkX project and found it interesting to work with as I am already interested in python and Git. Looking forward to your guidance.

**Comment by ManasviEmmadi** [comment]
Hello @MridulS, I am Manasvi, and have really good experience with Python Programming and Data Analytics. I have been approved for the contribution round of outreachy internship program. I have submitted the result for task 1, and I look forward to delving deeper into networkx!

**Comment by Qudirah** [comment]
Hello. I am Qudirah.

**Comment by PurviChaurasia** [comment]
Hi @MridulS @paulitapb I am Purvi Chaurasia, A CS student from New Delhi, India. I would like to contribute NetworkX for Outreachy 2023. Since the project revolves around creating interactive notebooks, after checking out [NetworkX Algorithms](https://networkx.org/nx-guides/content/algorithms/index.html) I felt like there should be an introductory notebook which covers the basic functions and terminologies of graphs using NetworkX. Can I start working on creating such a notebook? Would love to get your opinion on it :)

**Reply by MridulS** [reply]
> which covers the basic functions and terminologies of graphs using NetworkX. 

Could you explain a bit more about this? What do you plan on covering here?

Thanks!

**Reply by PurviChaurasia** [reply]
> > which covers the basic functions and terminologies of graphs using NetworkX.
> 
> Could you explain a bit more about this? What do you plan on covering here?
> 
> Thanks!

Perhaps just a starter notebook covering the following things with fun and relevant visualization -

- What is a graph?
- Terminologies such as:

1. Order
2. Size
3. Degree - indegree, outdegree
4. Different Types of Graph

I can add a lot others, this is just for a start.

**Reply by MridulS** [reply]
IMO that would end up duplicating a lot of the tutorial https://networkx.org/documentation/latest/tutorial.html
But please feel free to suggest anything you think is missing from our tutorial :)

The notebooks in nx-guides are usually for "what to do once I know the basics of graphs".

**Reply by PurviChaurasia** [reply]
Oh okay, yes just checked that out, extremely sorry for the confusion!

**Reply by PurviChaurasia** [reply]
Had another suggestion on the algorithm side for the notebooks, as far as I know there aren't any covering shortest path right? - Dijkstra and Bellman Ford?

**Reply by PurviChaurasia** [reply]
If there aren't any, perhaps I can start working on that @MridulS

**Reply by MridulS** [reply]
We don't really expect you to start working on the notebooks in the contribution phase (you are welcome to if you want to!) but first priority should be to go through all the tasks in the outreachy repository.

**Reply by PurviChaurasia** [reply]
Okay, sure. Thanks!

**Comment by mimansa111** [comment]
Hello everyone, I am Sai Mimansa Gupta, a pre-final undergraduate student at IIT Roorkee. I have decent experience in Python Programming and I am entirely new to Open-source. I look forward to working with you all!

**Comment by MridulS** [comment]
Hello everyone! Super excited to see everyone. We also have a discord server available, so if you need to ask some questions in a chat like interface you can use discord too. But please only use the "networkx" channel, we share this discord server with multiple other projects and we should be mindful of that :)
You can look at the networkx channel on the left side of your screen.

<img width="1404" alt="Screenshot 2023-03-08 at 7 27 01 PM" src="https://user-images.githubusercontent.com/5363860/223755300-e661484b-61ae-4e2e-b95e-afe6ec55ef50.png">

You don't *need* to join discord, it's perfectly fine to keep on using github discussions if you are stuck or have any questions.

You can join the discord server with this link: https://discord.com/invite/vur45CbwMz

Direct link to "networkx" channel (this only works once you have signed up on the Scientific Python Discord server): https://discord.com/channels/786703927705862175/1020054798357254255

**Comment by Hawawou** [comment]
Hi all, my name is Hawawou and I'm excited to contribute to Networkx as part of my outreachy application

**Comment by mmbogajemimah** [comment]
Hi everyone. My name is Jemimah Mmboga I am really happy to contribute to Networkx as an outreachy applicant. I am happy to be starting out my open source journey with Networkx.

**Comment by Richiio** [comment]
Hello everyone! My name is Sarima Chiorlu. I am a prefinal student of systems engineering. I am happy to be here and can't wait to start contributing to Networkx

**Comment by PurviChaurasia** [comment]
@MridulS While going through the `examples/algorithms` folder, I found that it is lacking a lot of algorithms. Let me know if I'm wrong but these examples are used for better understanding of all algorithms implemented in NetworkX, right? If so, shouldn't more be incoporated?

**Reply by MridulS** [reply]
The examples folder is used to generate https://networkx.org/documentation/latest/auto_examples/index.html, and yes it would be nice to incorporate more of them, but the gallery examples are supposed to be short snippets with preferably pretty figures :)

If you have any ideas please do open a pull request!

**Reply by PurviChaurasia** [reply]
Sure, will do. Thanks!

**Comment by Jess-chege** [comment]
Hi everyone I'm Jacinta Chege an outreachy intern and having knowledge in **Python Programming, Git, Github, Jupyter Notebook and Data analysis**. I have been approved for project contributions and I'm kindly requesting to join the NetworkX project with my already attained skills and your guidance I'll be able to complete my contribution.

**Comment by PurviChaurasia** [comment]
Hi @MridulS @paulitapb  I wanted to know when should we start with our final application and how should we go about it? A little advice from your end would be extremely helpful!

**Reply by MridulS** [reply]
Please start it ASAP, don't wait till the last minute.

**Reply by PurviChaurasia** [reply]
Okay, thanks. Could I mail you regarding the same in order to discuss my further steps?

**Reply by MridulS** [reply]
Sure please do. But if it's something that doesn't *need* to be an email we prefer to keep the communication public as it helps everyone else too if they have the same questions. Plus you may get a response from someone else in the community quicker to your queries.

**Comment by Qudirah** [comment]
@MridulS I want to start too but my pull requests are not being attended to.

**Reply by MridulS** [reply]
There is no need to wait for getting the pull request merged/reviewed. Please start the application ASAP.

**Reply by Qudirah** [reply]
Alright.

**Comment by PurviChaurasia** [comment]
@MridulS @paulitapb I had a few general questions regarding the final applications - 

1. Do we have to add the PRs that we made for the outreachy repo as well? (the 5 initial tasks that we were asked to do)
2. Are issues to be marked as a seperate contribution than the PRs?
3. Can discussions that we initiated/ took part in be counted as contribution and should they be added?
4. What all is expected to be covered in the final timeline? A more clearer picture from your side will perhaps help all the applicants :)

**Reply by navyagarwal** [reply]
Yeah, we have to add the outreachy repo PRs as well, the same question was asked by Qudirah in the last round, check [this](https://github.com/networkx/networkx/discussions/6016#discussioncomment-4017764) out

**Reply by PurviChaurasia** [reply]
Yep that clears it! I also found a rough timeline to follow from the previous year discussions - [here](https://github.com/networkx/networkx/discussions/6016#discussioncomment-3980160)
Thanks a lot @navyagarwal !

**Reply by MridulS** [reply]
You can also add all the relevant links to the nx_pull_requests.txt file in the outreachy repo. Yes it says only pull requests but feel free to add links to issues/discussions.

**Reply by PurviChaurasia** [reply]
Yeah so do those issues and discussions need to be added as contributions on the main outreachy page as well?

**Reply by PurviChaurasia** [reply]
Also is NetworkX taking part in GSoC as well? Because I can't find it on the main orgs page. Would you recommend applying through both?

**Reply by navyagarwal** [reply]
> Also is NetworkX taking part in GSoC as well? Because I can't find it on the main orgs page.

Yeah, it's under the umbrella organization NumFOCUS

**Reply by MridulS** [reply]
> Would you recommend applying through both?

Yes please, it gives us (maintainers) a lot of leeway in allocating resources.

**Reply by devotuoma** [reply]
Good morning @ MridulS from Kenya. Could you just give me a hint of how many contributions I can record in the final application and the vital things to include please? Then if one is selected as an intern how does the organization gives a stipend or you may be selected as an intern and ends up not getting the stipend due to below-average work.

**Reply by Qudirah** [reply]
A rough timeline for the pedagogical notebooks project should follow something like
Week of May 29: Research about the algorithm, dataset or method for the pedagogical notebook 1
 Week of June 5: Implementation and Review 
Week of June 12: Review
Week of June 19: Research about the algorithm, dataset or method for the pedagogical notebook 2
 Week of June 26: Implementation and Review 
Week of July 3: Review
Week of July 10: Research about the algorithm, dataset or method for the pedagogical notebook 3 
Week of July 17: Implementation and Review 
Week of July 24: Review
Week of July 31: Research about the algorithm, dataset or method for the pedagogical notebook 4
 Week of Aug 7: Implementation and Review 
Week of Aug 14: Review
 Week of Aug 21: Wrap up of the internship

This is the time commitment from last round but adjusted to the current date.

**Reply by Qudirah** [reply]
> Good morning @ MridulS from Kenya. Could you just give me a hint of how many contributions I can record in the final application and the vital things to include please? Then if one is selected as an intern how does the organization gives a stipend or you may be selected as an intern and ends up not getting the stipend due to below-average work.

You should record as many contributions as you have made and the whole stipend will be handled by outreachy when time is right.

**Reply by PurviChaurasia** [reply]
Thanks a lot for the timeline! @Qudirah

**Reply by devotuoma** [reply]
Thank you so much @Qudirah

**Reply by Qudirah** [reply]
You are welcome :) @PurviChaurasia @devotuoma

**Comment by devotuoma** [comment]
@MridulS is there any additional questions we need to answer while making final application?

**Reply by MridulS** [reply]
No we don't have any additional questions.

**Reply by devotuoma** [reply]
Thank you.

**Reply by devotuoma** [reply]
How will the contributions be judged?

**Reply by MridulS** [reply]
We look at your application, your contributions and your previous experience for a holistic overview. There is no exact formula for this :)
